### PR TITLE
(#1039) Replaced @Rule with Throws matcher

### LINKED ISSUE #1039: TextTest.java:37-39: Replace all occurrences of @Rule...
The puzzle `1023-be859eed` from #1023 has to be resolved: 

https://github.com/yegor256/cactoos/blob/fb6a875013688413374f0c15ebbfe2d0753013b4/src/test/java/org/cactoos/TextTest.java#L37-L39

The puzzle was created by @rultor on 23-Jan-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by atapin**
It updates the `TextTest` by using `Throws` matcher instead of `@Rule`-marked exception check.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1100](https://github.com/yegor256/cactoos/issues/1100) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1100](https://github.com/yegor256/cactoos/issues/1100) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/da4e20df-9a6b-4ff1-9026-e5a288edbb12); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/ScalarTest.java [hunk 1]
```diff
@@ -35,24 +35,28 @@
...
-    @Rule
-    public final ExpectedException cause = ExpectedException.none();
-
     @Test
-    public void failForNullArgument() throws Exception {
-        this.cause.expect(IllegalArgumentException.class);
-        this.cause.expectMessage("NULL instead of a valid scalar");
-        new NoNulls<>(null).value();
+    public void failForNullArgument() {
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/ScalarTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/ScalarTest.java [hunk 2]
```diff
@@ -35,24 +35,28 @@
...
+        );
     }
 
     @Test
-    public void failForNullResult() throws Exception {
-        this.cause.expect(IllegalStateException.class);
-        this.cause.expectMessage("NULL instead of a valid value");
-        new NoNulls<>(() -> null).value();
+    public void failForNullResult() {
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/ScalarTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/TextTest.java [hunk 3]
```diff
@@ -26,41 +26,39 @@
...
-    @Rule
-    public final ExpectedException cause = ExpectedException.none();
-
     @Test
-    public void failForNullArgument() throws Exception {
-        this.cause.expect(IllegalArgumentException.class);
-        this.cause.expectMessage("NULL instead of a valid text");
-        new NoNulls(null).asString();
+    public void failForNullArgument() {
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/TextTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/TextTest.java [hunk 4]
```diff
@@ -26,41 +26,39 @@
...
 
     @Test
-    public void failForNullResult() throws Exception {
-        this.cause.expect(IllegalStateException.class);
-        this.cause.expectMessage("NULL instead of a valid result string");
-        new NoNulls(
-            () -> null
-        ).asString();
+    public void failForNullResult() {
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/TextTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/collection/NoNullsTest.java [hunk 5]
```diff
@@ -38,42 +38,45 @@
...
-     */
-    @Rule
-    public final ExpectedException exception = ExpectedException.none();
-
     @Test
     public void throwsErrorIfNullInToArray() {
-        this.exception.expect(IllegalStateException.class);
-        this.exception.expectMessage(
-            "Item #1 of #toArray() is NULL"
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/collection/NoNullsTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/collection/NoNullsTest.java [hunk 6]
```diff
@@ -38,42 +38,45 @@
...
-            new CollectionOf<>(1, null, 3)
-        ).toArray();
     }
 
     @Test
     public void throwsErrorIfNullInToArrayWithArg() {
-        this.exception.expect(IllegalStateException.class);
-        this.exception.expectMessage(
-            "Item #1 of #toArray(array) is NULL"
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/collection/NoNullsTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/collection/NoNullsTest.java [hunk 7]
```diff
@@ -38,42 +38,45 @@
...
-            new CollectionOf<>(1, null, 3)
-        ).toArray(new Object[3]);
     }
 
     @Test
     public void throwsErrorIfNullInContainsArg() {
-        this.exception.expect(IllegalArgumentException.class);
-        this.exception.expectMessage(
-            "Argument of #contains(T) is NULL"
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/collection/NoNullsTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/iterator/NoNullsTest.java [hunk 8]
```diff
@@ -48,27 +49,39 @@
...
-                "Item #0 of org.cactoos.iterator.NoNullsTest"
-            )
-        );
-        new NoNulls<>(
-            new Iterator<Integer>() {
-                @Override
-                public boolean hasNext() {
-                    return true;
-                }
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterator/NoNullsTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/iterator/NoNullsTest.java [hunk 9]
```diff
@@ -48,27 +49,39 @@
...
+                "Item #0 of Iterator@NoNullsTest is NULL",
+                IllegalStateException.class
+            )
+        );
     }
 
+    /*
+    * @todo #1039:15min Currently it's impossible to match error messages
+    *  by a pattern or partially. Replace {@link Rule} with {@link Throws} after
+    *  <a href="https://github.com/llorllale/cactoos-matchers/issues/108">llorllale/cactoos-matchers#108</a>
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterator/NoNullsTest.java]: @atapin it would also make sense to open a ticket so that `toString` is implemented on `Immutable`. You should do it and get some more points :)

File: src/test/java/org/cactoos/map/MapEnvelopeTest.java [hunk 10]
```diff
@@ -46,93 +43,73 @@
...
-    @Rule
-    public final ExpectedException exception = ExpectedException.none();
-
     @Test
     public void putThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#put() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/map/MapEnvelopeTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/map/MapEnvelopeTest.java [hunk 11]
```diff
@@ -46,93 +43,73 @@
...
         );
     }
 
     @Test
     public void removeThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#remove() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/map/MapEnvelopeTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/map/MapEnvelopeTest.java [hunk 12]
```diff
@@ -46,93 +43,73 @@
...
         );
     }
 
     @Test
     public void putAllThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#putAll() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/map/MapEnvelopeTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/map/MapEnvelopeTest.java [hunk 13]
```diff
@@ -46,93 +43,73 @@
...
         );
     }
 
     @Test
     public void clearThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#clear() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/map/MapEnvelopeTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/text/StrictTest.java [hunk 14]
```diff
@@ -40,21 +38,20 @@
...
      * Ensures that Strict is failing on a negative predicate result.
      * @throws Exception If fails
      */
     @Test
-    public void failsIfPredicateIsNegative() throws Exception {
-        this.exception.expect(IllegalArgumentException.class);
-        this.exception.expectMessage(new StringContains("text"));
-        new Strict(s -> false, new TextOf("text")).asString();
+    public void failsIfPredicateIsNegative() {
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/text/StrictTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/text/StrictTest.java [hunk 15]
```diff
@@ -75,13 +72,18 @@ public void returnsUnchangedIfPredicateIsPositive() {
...
     @Test
-    public void failsIfNotMatchedWithPattern() throws Exception {
-        this.exception.expect(IllegalArgumentException.class);
-        this.exception.expectMessage("text");
-        new Strict(
-            Pattern.compile("^[a-zA-Z]+$"),
-            new TextOf("text12")
-        ).asString();
+    public void failsIfNotMatchedWithPattern() {
+        MatcherAssert.assertThat(
```
Comment (victornoel) [inline src/test/java/org/cactoos/text/StrictTest.java]: @atapin please use `Assertion` instead of the static methods from `MatcherAssert`

File: src/test/java/org/cactoos/iterator/NoNullsTest.java (hunk 9 above)
Comment (atapin) [reply src/test/java/org/cactoos/iterator/NoNullsTest.java]: @victornoel thanks, good point! The issue is here #1102

**Comment by victornoel** [comment]
@atapin thanks

@llorllale seems good to merge. Not sure why appveyor fails, it is not related to this PR apparently…

**Comment by llorllale** [comment]
@atapin remove github keywords from the PR description (see our PR [template](https://github.com/yegor256/cactoos/blob/master/.github/PULL_REQUEST_TEMPLATE.md))

**Comment by atapin** [comment]
@llorllale done

**Comment by llorllale** [comment]
@atapin there are merge conflicts

**Comment by atapin** [comment]
@llorllale fixed

File: src/test/java/org/cactoos/map/MapEnvelopeTest.java [hunk 16]
```diff
@@ -42,98 +40,82 @@
  * @checkstyle JavadocMethodCheck (500 lines)
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  * @checkstyle DiamondOperatorCheck (500 lines)
+ * @todo #1039:15min Replace all invocations of the
+ *  {@link MatcherAssert#assertThat(String, Object, Matcher)}
```
Comment (llorllale) [inline src/test/java/org/cactoos/map/MapEnvelopeTest.java]: @atapin can you rephrase and say "all overloads of `MatcherAssert#assertThat`?

**Comment by llorllale** [comment]
@atapin just one more minor comment :)

**Comment by atapin** [comment]
@llorllale resolved

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#1100](https://github.com/yegor256/cactoos/issues/1100) is not in the agenda of @victornoel[/z](https://www.0crat.com/u/victornoel), can't inspect

**Comment by 0crat** [comment]
The job [#1100](https://github.com/yegor256/cactoos/issues/1100) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)
